### PR TITLE
use code_blockt::statements() where appropriate

### PR DESCRIPTION
**Pull request by kroening**
add/move are type-safe; copy_to_operands and move_to_operands are not.

### CODE REVIEW

**Comment by kroening** [comment]
I'll clang-format once reviewed; it's easier to review without.

File: src/util/std_code.h [hunk 1]
```diff
@@ -118,12 +118,12 @@ class code_blockt:public codet
 
   void move(codet &code)
   {
-    move_to_operands(code);
+    exprt::move_to_operands(code);
```
Comment (thk123) [inline src/util/std_code.h]: What's the reason for this change?

File: src/util/std_code.h [hunk 2]
```diff
@@ -101,21 +101,39 @@ inline codet &to_code(exprt &expr)
 class code_blockt:public codet
 {
 public:
+  typedef std::vector<codet> code_operandst;
+
+  code_operandst &code_operands()
```
Comment (thk123) [inline src/util/std_code.h]: :question: What's the purpose of having both `code_operands()` and `statements()`?

**Review by thk123 — COMMENTED** [review]
This would I suppose be a good time to apply this change in TG as well - how urgently do you want to get this in? It still needs a bump and I'd rather not see the introduction of synonym methods (labelled :question: )

File: src/util/std_code.h (hunk 2 above)
Comment (kroening) [reply src/util/std_code.h]: The idea is to eventually have code_operands() in codet, once codet stops inheriting from exprt.

**Comment by kroening** [comment]
No genuine urgency to do this on TG.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 658c93f3bb130e7997a0b357097750eaadf989b1).
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 3]
```diff
@@ -787,16 +787,16 @@ code_blockt &java_bytecode_convert_methodt::get_or_create_block_for_pcrange(
     return this_block;
 
   // Find the child code_blockt where the queried range begins:
-  auto child_iter=this_block.operands().begin();
+  auto child_iter=this_block.statements().begin();
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: Throughout this commit: please weave in clang-format style spacing while touching the various lines.

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 4]
```diff
@@ -787,16 +787,16 @@ code_blockt &java_bytecode_convert_methodt::get_or_create_block_for_pcrange(
...
   // Skip any top-of-block declarations;
   // all other children are labelled subblocks.
-  while(child_iter!=this_block.operands().end() &&
-        to_code(*child_iter).get_statement()==ID_decl)
+  while(child_iter!=this_block.statements().end() &&
+        child_iter->get_statement()==ID_decl)
     ++child_iter;
-  assert(child_iter!=this_block.operands().end());
+  assert(child_iter!=this_block.statements().end());
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: Should be a `DATA_INVARIANT`

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 5]
```diff
@@ -787,16 +787,16 @@ code_blockt &java_bytecode_convert_methodt::get_or_create_block_for_pcrange(
...
   std::advance(child_iter, child_offset);
-  assert(child_iter!=this_block.operands().end());
-  auto &child_label=to_code_label(to_code(*child_iter));
+  assert(child_iter!=this_block.statements().end());
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: This `assert` actually never was good enough as far as I can tell: the iterator should have been advanced well beyond the `end()` iterator.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: b8d45ef6683f90f19232647ff55083bb17dd747b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86391667

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 0dd049c726b6e93a471d697fb271500a445b142c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86663001

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 2311e72f5e7901ec748c85eff0af71a8a0c80766).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87142010

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 6]
```diff
@@ -1770,7 +1770,7 @@ codet java_bytecode_convert_methodt::convert_instructions(
           }
           else
             for(const auto &statement : more_code.statements())
-              c.copy_to_operands(statement);
+              to_code_block(c).add(statement);
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: Use `to_code_block(c).append(more_code);`?

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 361bba3376b41973e72d599e9bed9c5583feb50b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87625389
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp (hunk 6 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: Indeed, done

File: src/goto-instrument/goto_program2code.cpp [hunk 7]
```diff
@@ -1681,12 +1681,9 @@ static bool has_labels(const codet &code)
   return false;
 }
 
-static bool move_label_ifthenelse(
-    exprt &expr,
-    exprt &label_dest)
+static bool move_label_ifthenelse(exprt &expr, exprt &label_dest)
```
Comment (smowton) [inline src/goto-instrument/goto_program2code.cpp:1684]: Unrelated changes?

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 7813fecab6e33da29797b9d0b2f8f91c7ceba727).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87882190

File: src/goto-instrument/goto_program2code.cpp (hunk 7 above)
Comment (kroening) [reply src/goto-instrument/goto_program2code.cpp:1684]: That was clang-format.

**Review by martin-cs — APPROVED** [review]
goto-instrument changes look fine.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: ec1b59ad887d3441a048068f8f80c17780532107).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88419397
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 5dceb25afd7d1dbb6701c93ba2346053abf9722c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88420660
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 01fd831977b6fa1d9ff9fd017de9c5ec2ef958a4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88424313
